addons/account_payment/models/account_move.py

```diff
@@ -69,6 +70,35 @@ def _has_to_be_paid(self):
             and not pending_transactions
         )
 
+    def _get_online_payment_error(self):
+        """
+        Returns the appropriate error message to be displayed if _has_to_be_paid() method returns False.
+        """
+        self.ensure_one()
+        transactions = self.transaction_ids.filtered(lambda tx: tx.state in ('pending', 'authorized', 'done'))
+        pending_transactions = transactions.filtered(
+            lambda tx: tx.state in {'pending', 'authorized'}
+                       and tx.provider_code not in {'none', 'custom'})
+        enabled_feature = str2bool(
+            self.env['ir.config_parameter'].sudo().get_param(
+                'account_payment.enable_portal_payment'
+            )
+        )
+        errors = []
+        if not enabled_feature:
+            errors.append(_("This invoice cannot be paid online."))
+        if not (self.amount_residual or not transactions and not self.currency_id.is_zero(self.amount_residual) and self.amount_total):
```

Comment:
If there is a `amount_residual`, there is a `amount_total`...
`self.amount_residual` and `not self.currency_id.is_zero(self.amount_residual)` are redundant.
Is this condition not enough:
```suggestion
        if transactions or self.currency_id.is_zero(self.amount_residual):
```